src/Nethermind/Nethermind.Shutter/ShutterP2P.cs

```diff
@@ -102,11 +101,13 @@ public ShutterP2P(IShutterConfig shutterConfig, ILogManager logManager, IFileSys
     public async Task Start(Multiaddress[] bootnodeP2PAddresses, Func<Dto.DecryptionKeys, Task> onKeysReceived, CancellationTokenSource? cts = null)
     {
         _cts = cts ?? new();
-        _ = _router!.RunAsync(_peer, token: _cts.Token);
-        _ = _disc.DiscoverAsync(_peer.Address, _cts.Token);
+
+        await _peer.StartListenAsync([_address], _cts.Token);
+        await _router.StartAsync(_peer, _cts.Token);
+        _ = _disc.DiscoverAsync([Multiaddress.Decode(_address)], _cts.Token);
         _peerStore.Discover(bootnodeP2PAddresses);
```

Comment:
`Discover` is designed to receive multiple p2p addresses of the same peer. You need to discover bootnodeP2PAddresses one by one here looks like